lib/Common/CommonBasic.h

```diff
@@ -8,6 +8,16 @@
 #include "CommonDefines.h"
 #define _CRT_RAND_S         // Enable rand_s in the CRT
 
+#pragma warning(push)
+#if defined(PROFILE_RECYCLER_ALLOC) || defined(HEAP_TRACK_ALLOC) || defined(ENABLE_DEBUG_CONFIG_OPTIONS)
+#ifdef __clang__
+namespace std { struct type_info; }
+typedef std::type_info type_info;
```

Comment:
Is "std::type_info" declared in "<typeinfo>"? If so may "using std::type_info" after "#include<typeinfo>".